src/code/index.js

```diff
@@ -88,4 +88,38 @@ function previewLink(button) {
     console.log("Placeholder value:", inputElement.placeholder);
     console.log("Input value:", inputValue);
     // Rest of the code...
-}
\ No newline at end of file
+}
+
+//Function to check if a string is a valid URL
+function isValidURL(url) {
+    try {
+        new URL(url);
+        return true;
+    } catch (error) {
+        return false;
+    }
+}
+
+//Function to handle input change event
+function handleInputChange() {
+    const linkInput = document.getElementById('linkInput');
+    const errorText = document.getElementById('errorText');
+    const previewButton = document.getElementById('previewButton');
```

Comment:
Same, these IDs are not defined in linkInput, what you can do is modify your function as such that it takes the input element and preview button or the `link-box` container and apply functionality accordingly.